### PR TITLE
feat(metrics): add parse rate info to app scan data

### PR DESCRIPTION
**Pull request by kopecs**
PR checklist:

- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] `changelog.d/<issue>.<type>` is a file with the _what_, _why_, and _how_ of the change. (See #5815)
  - \<issue> is `pa-312` (Linear ticket), `gh-1234` (GitHub issue), or `new-gizmo` (unique semantic name)
  - \<type> is `added`, `changed`, `fixed`, or `infra`.
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure on any of this, please see:

- [Changelog guidelines](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry)
- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

File: PRIVACY.md [hunk 1]
```diff
@@ -322,6 +322,7 @@ The classes of scan data are:
 - Review and review-requester identifying data (e.g., pull-request ID, branch, merge base, request author)
 - Scan metadata, including type of scan and scan parameters (e.g., paths scanned and extensions of ignored files)
 - Timing metrics (e.g., time taken to scan per-rule and per-path)
+- Parse metrics (e.g., number of files targed and parsed per-language)
```
Comment (ajbt200128) [inline PRIVACY.md]:
```suggestion
- Parse metrics (e.g., number of files targeted and parsed per-language)
```

File: cli/src/semgrep/app/scans.py [hunk 2]
```diff
@@ -213,6 +215,15 @@ def report_findings(
                 "errors": [error.to_dict() for error in errors],
                 "total_time": total_time,
                 "unsupported_exts": dict(ignored_ext_freqs),
+                "parse_rate": {
+                    lang: {
+                        "targets_parsed": data.num_targets - data.targets_with_errors,
```
Comment (ajbt200128) [inline cli/src/semgrep/app/scans.py:220]: Nit: `targets_parsed` could end up negative which could mess up data. 
```suggestion
                        "targets_parsed": min(data.num_targets - data.targets_with_errors,0),
```

**Review by ajbt200128 — APPROVED** [review]
Left two nits but looks good

File: cli/src/semgrep/app/scans.py (hunk 2 above)
Comment (kopecs) [reply cli/src/semgrep/app/scans.py:220]: `targets_with_errors` is necessarily less by construction, so I don't think adding an extra check here to hide a failure is the right choice.

**Pull request merged**